packages/@expo/cli/src/start/server/metro/runServer-fork.ts

```diff
@@ -95,6 +96,23 @@ export const runServer = async (
     end();
   });
 
+  // Extend the close method to ensure all websocket servers are closed, and connections are terminated
+  const originalClose = httpServer.close.bind(httpServer);
+
+  httpServer.close = function closeHttpServer(callback) {
```

Comment:
Do we control the code that calls `httpServer.close`? It would be better to put this new logic in a function called `stopServer(httpServer)` (for example) and call that instead of overwriting `httpServer.close` but I understand if we don't control that code.

Reply:
Yeah, I think we call it from this place here: https://github.com/expo/expo/blob/8ea118e9fd2576792ef6aab9fc49f2e9ed0d0340/packages/%40expo/cli/src/start/server/BundlerDevServer.ts#L334

Reply:
The idea is to self-contain the websocket implementation and not leak it out to other functions. This is an artifact from multiple bundlers, different platforms, and forks in how the server is started. We're actively reducing and unifying these codepaths but self-containing seems like the best way to mitigate regressions.